internal/component/otelcol/connector/connector.go

```diff
@@ -33,6 +33,7 @@ const (
 	ConnectorLogsToTraces
 	ConnectorLogsToMetrics
 	ConnectorLogsToLogs
+	ConnectorAnyToMetrics
```

Comment:
I don't mind having a `ConnectorAnyToMetrics`, but I think it would be cleaner to use a bit field like [this one](https://github.com/grafana/agent/blob/0e36aa27c95b7d600be1722ba80d56e8227d048b/internal/component/otelcol/exporter/exporter.go#L47-L72). This will also prevent us from needing a separate branch in the `switch` statement below for every permutation of inputs and outputs.

Having it as part of this `iota` makes it feels like it's a completely different connector type, which is also not very accurate.